### ISSUE TITLE
Opt-out of useless transforms 

### ISSUE THREAD

**Issue by davidbonnet** [alangpierce/sucrase#602]
Would it be considered to allow opting-out of certain transforms that are supported by the environment the code runs in?

For instance, optional chaining and nullish coalescing are both supported by Node 14. While the transform itself is blazingly fast, resulting in additional function definitions and calls deeply harm runtime performance:

Original code, fully supported by Node 14:
```js
const value = data?.property ?? {}
```

Transformed code:
```js
function _nullishCoalesce(lhs, rhsFn) {
  if (lhs != null) {
    return lhs;
  } else {
    return rhsFn();
  }
}
function _optionalChain(ops) {
  let lastAccessLHS = undefined;
  let value = ops[0];
  let i = 1;
  while (i < ops.length) {
    const op = ops[i];
    const fn = ops[i + 1];
    i += 2;
    if ((op === 'optionalAccess' || op === 'optionalCall') && value == null) {
      return undefined;
    }
    if (op === 'access' || op === 'optionalAccess') {
      lastAccessLHS = value;
      value = fn(value);
    } else if (op === 'call' || op === 'optionalCall') {
      value = fn((...args) => value.call(lastAccessLHS, ...args));
      lastAccessLHS = undefined;
    }
  }
  return value;
}
const value = _nullishCoalesce(_optionalChain([data, 'optionalAccess', _ => _.property]), () => ({}));
```

**Comment by alangpierce** [comment]
Thanks for reporting, I definitely agree, and I've been thinking about a Sucrase 4.0 release that drops these transforms entirely, or makes them opt-in.

I think the right approach is to add a single new flag `disableESTransforms` (or `disableJSTransforms`) that opts out of optional chaining, nullish coalescing, class fields, numeric separators, and optional catch binding. I think they're all supported as of node 14 and supported in latest browsers (except some class fields cases in Safari IIRC), but would be good to verify. I'd be happy to accept a PR to add that flag. I think the main code changes are to add the flag, use it in the `RootTransformer` constructor, and write some tests for the new behavior.

**Comment by alangpierce** [comment]
This is now fixed and released as 3.19.0. You can specify `disableESTransforms` in the Sucrase options to opt out of transforms for optional chaining, nullish coalescing, class fields, numeric separators, and optional catch binding.
